### PR TITLE
Protect against version downgrade in `secrets tune` command

### PR DESCRIPTION
**Pull request by catsby**
Fixed https://github.com/hashicorp/vault/issues/5237 by guarding against removing the `version` option in secrets backends. 

I'm not 100% confident in the placement of the test, I struggled constructing one in `logical_system_test.go` where maybe it makes more sense to be. 

The test is currently in `command/secrets_tune_test.go`, I was able to construct a test that reproduces the regression and verifies it's fixed by 27f008b .

### CODE REVIEW

File: vault/logical_system.go [hunk 1]
```diff
@@ -1342,6 +1343,16 @@ func (b *SystemBackend) handleTuneWriteCommon(ctx context.Context, path string,
 				resp = &logical.Response{}
 				resp.AddWarning(fmt.Sprintf("Upgrading mount from version %d to version %d. This mount will be unavailable for a brief period and will resume service shortly.", meVersion, optVersion))
 			}
+		} else {
+			// if version is not included in the
```
Comment (briankassouf) [inline vault/logical_system.go]: This is a nit-pick, but the text width on this comment block seems off. It should be 80 chars no matter the indentation.

Comment (catsby) [reply vault/logical_system.go]: A good nit though, not sure how I let that by I'm normally pretty on-top of things like that 😄 
Fixed in 8f19ec962

**Comment by calvn** [comment]
This is semi-related since you're working with the options param: https://github.com/hashicorp/vault/blob/fb89af7cfa66b033951fa4c44ff31718537f094f/http/sys_mount_test.go#L701 and https://github.com/hashicorp/vault/blob/fb89af7cfa66b033951fa4c44ff31718537f094f/http/sys_mount_test.go#L706 should be removed since where's testing for the absence of the values after it's being unset. The current logic doesn't let you unset the options map once that's set since we are only writing to storage if `len(options) > 0`.

**Comment by catsby** [comment]
Closing in favor of https://github.com/hashicorp/vault/pull/5809

**Pull request closed without merging**